### PR TITLE
Add check for negative stripe index in ORC reader

### PR DESCRIPTION
**Pull request by vuule**
Fixes CI failure

### CODE REVIEW

**Review by cwharris — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

File: cpp/src/io/orc/aggregate_orc_metadata.cpp
Comment (bdice) [inline cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: I suggest that we could upcast the `auto=cudf::size_type` to match the `size_t` that the `size()` function is returning, rather than play type gymnastics by downcasting the result of `size()` (requiring a `decltype()`) and also checking `>= 0`? The end result appears to be shorter and safer. (Excuse the lack of clang-format)
```suggestion
        CUDF_EXPECTS(static_cast<std::size_t>(stripe_idx) < per_file_metadata[src_file_idx].ff.stripes.size(), "Invalid stripe index");
```

**Review by bdice — CHANGES_REQUESTED** [review]
See comment.

File: cpp/src/io/orc/aggregate_orc_metadata.cpp
Comment (vuule) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: but then we would be static casting to size_t a value that can be negative. IMO this option works "by accident", the explicit check here is preferable to me.

Comment (bdice) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: Can a stripe index be negative? A `size_type==int` can be negative but that doesn't mean that `stripe_idx` should ever take on negative values (unless I'm unaware of how it's used). In the same way, we would need to know in the current snippet that the `size()` call returning a `size_t` won't overflow a `size_type` when being cast. I would argue that the purported safety of _either option_ is conditional on prior knowledge of the values, and is not an intrinsic guarantee of the choice of type/casting.

Comment (vuule) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: Stripe index is a value passed by the user, so we need to check if it's in valid range. It is imposible for size to be more than max size_type, since it's always (way) smaller than the number of rows.

**Review by bdice — APPROVED** [review]
~Please feel free to merge this PR if it's fixing CI -- I don't to hold up CI for this suggestion so close to code freeze -- but let's discuss my suggestion further in the comment thread.~ edit: LGTM

File: cpp/src/io/orc/aggregate_orc_metadata.cpp
Comment (bdice) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: > Stripe index is a value passed by the user

Okay, that's what I was missing. I assumed it was a loop index or something where we had stronger guarantees about its potential values.

Comment (vuule) [reply cpp/src/io/orc/aggregate_orc_metadata.cpp:178]: Yeah, this check is just input validation, which is what the failing test is validating.

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**
